### PR TITLE
Added site_id and fixed versions for feature install.

### PR DESCRIPTION
**Pull request by jdelgado-ntent**
Our organization uses the site ID of each web service for logging sanity and have a need to have this set. I added the variable and the logic to add one if not supplied and verify it isn't in use.

Also, during my travels, while attempting to use the installation mechanic, I found the versions listed in the .pp files were reversed. IIS-* is used in older Windows Servers and Web-* in newer, not the other way around.

### CODE REVIEW

**Comment by igalic** [comment]
@jdelgado-ntent i would very much appreciate if you could squash & rebase your commits.
*****
after a modulesync… ping @jyaworski

**Comment by jdelgado-ntent** [comment]
@igalic I have squashed and rebased my commits. Thanks for the heads up. Don't know what else to do next after this.

**Comment by juniorsysadmin** [comment]
If you rebase to master and squash again the existing tests should now work.
Are you absolutely sure on the kernelmajversion changes? Also would be good to see some additional tests for this.

**Comment by jdelgado-ntent** [comment]
I am sure about the kernel versions. I have both a Windows 2008 R2 vm and a Server 2012 R2 vm and checked it out manually. 2008 uses IIS-*, 2012 uses Web-*

File: .rubocop.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ AllCops:
 
 # Configuration parameters: AllowURI, URISchemes.
 Metrics/LineLength:
-  Max: 328
+  Max: 417
```
Comment (jyaworski) [inline .rubocop.yml]: This will be overwritten by modulesync. Why is this needed?

**Comment by jdelgado-ntent** [comment]
The test for rubocop failed because "C: Line is too long. [416/328]" so I increased the line length to accommodate the longest line. (416 characters). Is this not the proper way to fix this?

File: .rubocop.yml (hunk 1 above)
Comment (jdelgado-ntent) [inline .rubocop.yml]: The test for rubocop failed because "C: Line is too long. [416/328]" so I increased the line length to accommodate the longest line. (416 characters). Is this not the proper way to fix this?

Comment (jyaworski) [inline .rubocop.yml]: Please split the line instead.

Comment (jdelgado-ntent) [inline .rubocop.yml]: Done.

**Comment by jyaworski** [comment]
Can you squash?

**Comment by jdelgado-ntent** [comment]
Squashed.

**Comment by jyaworski** [comment]
Now there are merge conflicts. Can you look into this?

**Comment by jdelgado-ntent** [comment]
Going to close pull request and recreate with the changes. Something is wrong with git and it keeps giving me errors trying to merge/rebase.
